src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6322,12 +6561,190 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
+                // Branch on the success parameter to the next block
+                Debug.Assert(resultBlock != null);
+                Debug.Assert(outParameterPlaceholder != null);
+                Debug.Assert(outParameterFlowCapture != -1);
+
+                // if (!outParameterFlowCapture) goto resultBlock;
+                ConditionalBranch(new FlowCaptureReferenceOperation(outParameterFlowCapture, outParameterPlaceholder.Syntax, outParameterPlaceholder.Type, constantValue: null), jumpIfTrue: false, resultBlock);
+                // else goto next block;
+                LeaveRegionsUpTo(resultRegion);
+                AppendNewBlock(new BasicBlockBuilder(BasicBlockKind.Block), linkToPrevious: true);
+            }
+            else
+            {
+                LeaveRegionsUpTo(resultRegion);
+            }
+
+            if (operation.HandlerCreationHasSuccessParameter)
+            {
+                RemovePlaceholder(outParameterPlaceholder!);
+            }
+
+            var appendCalls = collectAppendCalls(operation);
+
+            int appendCallsLength = appendCalls.Length;
+            for (var i = 0; i < appendCallsLength; i++)
+            {
+                var appendRegion = new RegionBuilder(ControlFlowRegionKind.LocalLifetime);
+                EnterRegion(appendRegion);
+                var appendCall = appendCalls[i];
+                if (operation.HandlerAppendCallsReturnBool)
+                {
+                    Debug.Assert(resultBlock != null);
+
+                    if (i == appendCallsLength - 1)
+                    {
+                        // No matter the result, we're going to the result block next. So just visit the statement, and if the current block can be
+                        // combined with the result block, the compaction machinery will take care of it
+                        VisitStatement(appendCall.AppendCall);
+                        AppendNewBlock(resultBlock, linkToPrevious: true);
+                    }
+                    else
+                    {
+                        // if (!appendCall()) goto result else goto next block
+                        ConditionalBranch(VisitRequired(appendCall.AppendCall), jumpIfTrue: false, resultBlock);
+                        AppendNewBlock(new BasicBlockBuilder(BasicBlockKind.Block), linkToPrevious: true);
+                    }
+                }
+                else
+                {
+                    VisitStatement(appendCall.AppendCall);
+
+                    if (i == appendCallsLength - 1 && operation.HandlerCreationHasSuccessParameter)
+                    {
+                        Debug.Assert(resultBlock != null);
+                        AppendNewBlock(resultBlock, linkToPrevious: true);
+                    }
+                }
+                LeaveRegionsUpTo(appendRegion);
+                LeaveRegion();
+            }
+
+            if (appendCallsLength == 0 && operation.HandlerCreationHasSuccessParameter)
```

Comment:
>appendCallsLength == 0 && operation.HandlerCreationHasSuccessParameter [](http://example.com/codeflow?start=16&length=70)

Manual management of basic blocks here feels unnecessary complicated. I think all we really need to do is to ```_currentBasicBlock = null;``` after conditional branches and here unconditionally append ```resultBlock``` if we have one.

Reply:
This doesn't look addressed. The suggestion isn't followed, there is stil manual management that is not related to branching.